File: playwright-typescript/pages/SettingsPage.ts
import {type Page} from '@playwright/test';
import {BasePage} from './factory/BasePage';

export class SettingsPage extends BasePage {
    readonly settingsText: string = 'settings-text';
    readonly view: string = 'view-settings';

    constructor(page: Page) {
        super(page);
    }

    async getSettingsText(): Promise<string> {
        return await this.getTestIdText(this.settingsText);
    }
}

File: selenium-python/pages/settings_page.py
from __future__ import annotations

from pages.base_page import BasePage


class SettingsPage(BasePage):
    settings_text = "settings-text"
    view = "view-settings"

    def get_settings_text(self) -> str:
        return self.get_test_id_text(self.settings_text)
